midterm-2023-11-06/code/5066-agon-jakupi/script.ts

```diff
@@ -0,0 +1,210 @@
+
+type RockBand = {
+    ID: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    genre: string;
+    members: string;
+    albums: number;
+};
+
+interface albums{ 
+name: string;
+
+}
+
+document.addEventListener("DOMContentLoaded", siteCode);
+let rockband : RockBand[] = [];
+async function siteCode() {
+    const data = await loadData();
+    rockband = data.metalBands;
+
+    fillLocation(rockband)
+    
+
+    const sortButton = document.getElementById("sort-id")!
+    sortButton.addEventListener("click", sortById)
+
+
+    const sortName = document.getElementById("sort-name")!
+    sortName.addEventListener("click", sortByName)
+
+    const applyName = document.getElementById("name-filter")
+    applyName?.addEventListener("input", applyNameFilter)
+
+ 
+
+    const applyButton = document.getElementById("apply-filter")!
+    applyButton.addEventListener("click", applyFilter)
+    populateTable(rockband)
+
+
+}
+
+// Function to populate the HTML table with rock band data
+
+const fillLocation = (rockband: RockBand[]) => {
+    const filter = document.getElementById("country-filter")!;
+
+    const countries = new Set<string>();
+    for (const bandrock of rockband) {
+        countries.add(bandrock.country);
+    }
+
+    for (const country of countries) {
+        const option = document.createElement("option");
+        option.value = country;
+        option.innerHTML = country;
+        filter.appendChild(option);
+    }
+}
+
+const CountryFind = (rockband: RockBand) => {
+    let bandLocation = rockband.location;
+    let Parts = bandLocation.split(",");
+    let LastBit = Parts[Parts.length - 1]
+    return LastBit
+
+}
+
+const applyFilter = () => {
+    const CountriesElement = document.getElementById("country-filter") as HTMLSelectElement;
```

Comment:
Aside from the naming, the format of the method is ok. However, since the only selection is `undefined` the filter is not operational +10